core/src/main/java/nu/yona/server/subscriptions/service/UserUpdateService.java

```diff
@@ -143,7 +143,7 @@ public User updateUser(UUID id, Consumer<User> updateAction)
 	{
 		// We add a lock here to prevent concurrent user updates.
 		// The lock is per user, so concurrency is not an issue.
-		return userRetrievalService.withLockOnUser(id, user -> {
+		return userLookupService.withLockOnUser(id, user -> {
```

Comment:
Why is `withLockOnUser` part of `UserLookupService`? I would say it should be part of `UserUpdateService`

Reply:
For reasons of dependencies. The method is called from the ``UserLookupService``, so moving this method to ``UserUpdateService`` would cause a two-way dependency between these two.